code/controllers/subsystem/vote.dm

```diff
@@ -234,97 +231,74 @@ SUBSYSTEM_DEF(vote)
 		return TRUE
 	return FALSE
 
-/datum/controller/subsystem/vote/proc/interface(client/C)
-	if(!C)
+/mob/verb/vote()
+	set category = "OOC"
+	set name = "Vote"
+	SSvote.ui_interact(usr)
+
+/datum/controller/subsystem/vote/ui_state()
+	return GLOB.always_state
+
+/datum/controller/subsystem/vote/ui_interact(mob/user, datum/tgui/ui)
+	// Tracks who is voting
+	if(!(user.client?.ckey in voting))
+		voting += user.client?.ckey
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Vote")
+		ui.open()
+
+/datum/controller/subsystem/vote/ui_data(mob/user)
+	var/list/data = list(
+		"mode" = mode,
+		"voted" = voted,
+		"voting" = voting,
+		"choices" = list(),
+		"question" = question,
+		"initiator" = initiator,
+		"started_time" = started_time,
+		"time_remaining" = time_remaining,
```

Comment:
Why is this sent to the client? If you provide the start time, just provide the time a vote will take, and predict it all on the client.

Reply:
time remaining is used in vote.dm to close the vote panel when it reaches 0. i can remove it but i'd need another way to close the panel.

Reply:
Are you sure this is an optimal solution? Will this cause an issue if there's time dilation?

Reply:
If you want perfect timings, you could use process to check that `world.time - started_time` has gone long enough, that way even in high time dilation it'll take the same amount of time.